sourcegraph/learn
# Add pre-commit hook

### What should this PR do?
Resolves [DEVED-35](https://sourcegraph.atlassian.net/browse/DEVED-35) by adding a pre-commit hook to ensure that any proposed changes in a commit pass our linting checks before being added.

### Why are we making this change?
Automating checks for code quality will help us ensure that we "do no harm" as we add new code to the project.

### What are the acceptance criteria? 
- Pre-commit hook should work for `tsx` and `ts` files.
- Instructions in the README are clear and comprehensible.

### How should this PR be tested?
- Check out the branch, and follow the config instructions in the README (under `Configure the pre-commit Githook`).
- Make some incorrect changes to `tsx` and `ts` files.
- Stage the files and attempt to commit them. Check that you cannot with errors in place.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
